src/resources/classes/timeSeries.ts

```diff
@@ -1,53 +1,44 @@
 // Copyright 2019 Cognite AS
 import { CogniteClient } from '../..';
 import {
+  CogniteExternalId,
+  CogniteInternalId,
   DatapointsMultiQuery,
   GetTimeSeriesMetadataDTO,
   LatestDataPropertyFilter,
+  Metadata,
 } from '../../types/types';
 import { BaseResource } from './baseResource';
 
 export class TimeSeries extends BaseResource<GetTimeSeriesMetadataDTO>
   implements GetTimeSeriesMetadataDTO {
-  public get externalId() {
-    return this.props.externalId;
-  }
-  public get name() {
-    return this.props.name;
-  }
-  public get isString() {
-    return this.props.isString;
-  }
-  public get metadata() {
-    return this.props.metadata;
-  }
-  public get unit() {
-    return this.props.unit;
-  }
-  public get assetId() {
-    return this.props.assetId;
-  }
-  public get isStep() {
-    return this.props.isStep;
-  }
-  public get description() {
-    return this.props.description;
-  }
-  public get securityCategories() {
-    return this.props.securityCategories;
-  }
-  public get createdTime() {
-    return this.props.createdTime;
-  }
-  public get lastUpdatedTime() {
-    return this.props.lastUpdatedTime;
-  }
-  public get id() {
-    return this.props.id;
-  }
+  public externalId?: CogniteExternalId;
+  public name?: string;
+  public isString: boolean;
+  public metadata?: Metadata;
+  public unit?: string;
+  public assetId?: CogniteInternalId;
+  public isStep: boolean;
+  public description: string;
+  public securityCategories?: number[];
+  public createdTime: Date;
+  public lastUpdatedTime: Date;
+  public id: CogniteInternalId;
 
   constructor(client: CogniteClient, props: GetTimeSeriesMetadataDTO) {
-    super(client, props);
+    super(client);
+    this.externalId = props.externalId;
```

Comment:
We could potentially add support in `BaseResource` for a list of prop names like `['externalId', 'name', 'isString', ...]`. And then use that list to create the `toJSON` return value and we could send the props to `BaseResource` as before.